pkg/omf/functions/index/omf.index.query.fish

```diff
@@ -0,0 +1,82 @@
+function omf.index.query -d 'Query packages in the index'
+  set -q XDG_CACHE_HOME
+    and set -l index_path "$XDG_CACHE_HOME/omf"
+    or set -l index_path "$HOME/.cache/omf"
+
+  set -lx q_type any
+  set -lx q_name ''
+  set -lx q_text ''
+
+  # Parse search terms.
+  for arg in $argv
+    switch "$arg"
+      case 'type:a*'
+        set q_type any
+      case 'type:p*'
+        set q_type package
+      case 'type:t*'
+        set q_type theme
+      case 'name:?*'
+        set -l IFS ':'
+        echo "$arg" | read dummy q_name
+      case 'text:?*'
+        set -l IFS ':'
+        echo "$arg" | read dummy q_text
+      case '*'
+        echo "Invalid search term: '$arg'" >&2
+        return 1
+    end
+  end
+
+  # Determine what files to search against based on type.
+  switch $q_type
+    case package
+      set packages $index_path/*/packages/*
+    case theme
+      set packages $index_path/*/themes/*
+    case '*'
+      set packages $index_path/*/{packages,themes}/*
+  end
+
+  # If there are no packages, there is nothing to search.
+  if not set -q packages[1]
+    return 1
+  end
+
+  # Perform a text search if any textual terms were given.
+  if test -n "$q_name" -o -n "$q_text"
+    set results (command awk '
+      BEGIN {
+        FS = "[ \t]*=[ \t]*";
+        q_name = tolower(ENVIRON["q_name"]);
+        q_text = tolower(ENVIRON["q_text"]);
```

Comment:
`-v` wasn't working how I needed at some point so I used environment variables instead. I don't remember what the problem was but seems to work fine now. `-v` is more explicit.